src/pu-utils.c

```diff
@@ -442,6 +442,62 @@ pu_wait_for_partitions(GError **error)
     return TRUE;
 }
 
+gboolean
+pu_set_hwreset(const gchar *device,
+               const gchar *hwreset,
+               GError **error)
+{
+    g_autofree gchar *cmd = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, 0);
+
+    if (g_strcmp0(hwreset, "") <= 0)
+        return TRUE;
+
+    if (g_strcmp0(hwreset, "enable") == 0) {
+        cmd = g_strdup_printf("mmc hwreset enable %s", device);
+    } else if (g_strcmp0(hwreset, "disable") == 0) {
+        cmd = g_strdup_printf("mmc hwreset disable %s", device);
+    } else {
+        g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
+                    "Invalid option for hwreset provided: %s", hwreset);
+        return FALSE;
+    }
+
+    if (!pu_spawn_command_line_sync(cmd, error)) {
+        if (strstr((*error)->message, "H/W Reset is already permanently")) {
+            g_clear_error(error);
+            g_debug("H/W Reset is already permanently set");
```

Comment:
You could parse `hwreset` here to get the correct word (enabled/disabled):
```suggestion
            g_debug("H/W Reset is already permanently %sd", hwreset);
```

Reply:
I don't think this will work since the state we want to set is not necessarily the one the eMMC is currently in.
What could work is using the error message as debug message.

Reply:
I see. What about we communicate it this way:
```c
g_debug("H/W reset already permanently set");
```
The error message, although being printed at debug level, may be confusing for users. It could be interpreted as an actual error, although it is not.